From 35f9300ae8f6ff3b2c5b32cf8c45a33b2075661e Mon Sep 17 00:00:00 2001
From: mj <mj@maxlinear.com>
Date: Fri, 9 May 2025 15:02:27 +0530
Subject: [PATCH] WLANRTSYS-88253 Allow set channel during CAC period from
 hostapd.

Issue : avoid set channel in hostapd , while running CAC.
Fix : FW already validing this. Hence, skipping this check in hostapd.
---
 hostapd/ctrl_iface.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index 0ad90c8df..5cd901592 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -3184,10 +3184,6 @@ static int hostapd_ctrl_iface_chan_switch(struct hostapd_iface *iface,
 		return -1;
 	}
 
-	if (!switch_type_zwdfs && iface->cac_started) {
-		wpa_printf(MSG_ERROR, "CAC is running, can't switch channel");
-		return -1;
-	}
 	/* Check if CCA/CSA in progress or CAC is running for MLO */
 	if (!mxl_hostapd_ctrl_iface_is_channel_allowed(iface)) {
 		wpa_printf(MSG_ERROR, "Channel switch isn't allowed for MLO vap");
-- 
2.43.0

